### PR TITLE
fix bufferd dynamo hatch 

### PR DESCRIPTION
**Pull request by botn365**
fix where a turbine with a buffer dynamo hatch makes more then 1A it wil not out put non complete amps in to the buffer #5061

### CODE REVIEW

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java [hunk 1]
```diff
@@ -575,8 +575,12 @@ public boolean addEnergyOutputMultipleDynamos(long aEU, boolean aAllowMixedVolta
...
-                    powerGain = aAmpsToInject > 0 ? aVoltage : aRemainder;
+                 for (int i = 0; i < Math.min(aDynamo.maxAmperesOut(), aAmpsToInject + 1); i++) {
+                    if (i == Math.min(aDynamo.maxAmperesOut(), aAmpsToInject)){
+                        powerGain = aRemainder;
+                    }else{
+                        powerGain =  aVoltage;
+                    }
                     aDynamo.getBaseMetaTileEntity().increaseStoredEnergyUnits(powerGain, false);
                     injected += powerGain;
                 }
```
Comment (Technus) [inline src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java]:
```java
int amps=Math.min(aDynamo.maxAmperesOut(), aAmpsToInject);
for (int i = 0; i < amps; i++) {
    aDynamo.getBaseMetaTileEntity().increaseStoredEnergyUnits(aVoltage, false);
}
injected+=aVoltage*amps;
if(aRemainder>0 && amps<aDynamo.maxAmperesOut()){
    aDynamo.getBaseMetaTileEntity().increaseStoredEnergyUnits(aRemainder, false);
    injected+=aRemainder;
}
```

Comment (Technus) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java]: This should work faster

Comment (Technus) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_MultiBlockBase.java]: someone should check if my change works, literally wrote it on the knee

**Comment by draknyte1** [comment]
It looks good at face value.

**Comment by Dream-Master** [comment]
@draknyte1 @bartimaeusnek  with @Technus changes or without ?

**Pull request merged**